apps/portals/src/portal-components/RouteControlWrapper.tsx

```diff
@@ -1,89 +1,117 @@
 import { SynapseComponent } from '../SynapseComponent'
-import { SynapseConfig } from '../types/portal-config'
-import { useEffect, useState } from 'react'
+import { ConfigRoute, NestedRoute } from '../types/portal-config'
 import { ArrowDropDown, ArrowDropUp } from '@mui/icons-material'
+import { Box, Typography, useMediaQuery, useTheme } from '@mui/material'
+import { RESPONSIVE_SIDE_PADDING } from '../utils'
 
-export type RouteControlWrapperProps = {
-  synapseConfig?: SynapseConfig
+export type RouteControlWrapperProps = NestedRoute & {
   // we have to pass in all the custom routes because unlike the home page the explore buttons configs aren't held in state
-  customRoutes: string[]
+  customRoutes: ConfigRoute[]
   searchParams?: any
 }
 
-type Props = RouteControlWrapperProps
-
 /**
  * RouteControl is the set of controls used on the /Explore page to navigate the
  * different keys.
- *
- * @param {*} { location, SynapseConfig, colors, history, customRoutes }
- * @returns
  */
-const RouteControlWrapper: React.FunctionComponent<Props> = ({
-  synapseConfig,
-  customRoutes = [],
-  searchParams,
-}) => {
+export default function RouteControlWrapper(props: RouteControlWrapperProps) {
+  const { customRoutes = [], searchParams } = props
   const location = useLocation()
   const history = useHistory()
   const pathname = location.pathname
   const subPath = pathname.substring('/Explore/'.length)
-  const handleChangesFn = (val: string, _index: number) => {
-    setSelectedTab(val.toUpperCase())
-    history.push(`/Explore/${val}`)
-  }
-  const routeControlProps: RouteControlProps = {
-    customRoutes,
-    handleChanges: handleChangesFn,
-    isSelected: (name: string) => name === subPath,
-  }
-  const [selectedTab, setSelectedTab] = useState<string>()
+  const handleChangesFn = useCallback(
+    (val: string, _index: number) => {
+      history.push(`/Explore/${val}`)
+    },
+    [history],
+  )
+  const routeControlProps: RouteControlProps = useMemo(
+    () => ({
+      customRoutes: customRoutes
+        .map((route) => route.path)
+        .filter(Boolean) as string[],
+      handleChanges: handleChangesFn,
+      isSelected: (name: string) => name === subPath,
+    }),
+    [customRoutes, handleChangesFn, subPath],
+  )
+
+  const selectedTab = subPath
   const [showSubNav, setShowSubNav] = useState<boolean>(false)
+  const theme = useTheme()
+  const isDesktopView = useMediaQuery(theme.breakpoints.up('sm'))
 
-  useEffect(() => {
-    setSelectedTab(subPath.toUpperCase())
-  }, [subPath])
+  const currentRoute = customRoutes.find((route) => route.path === selectedTab)
+  if (currentRoute) {
+    const pageName =
+      currentRoute.displayName ?? currentRoute.path?.replaceAll('/', '')
+    const documentTitle = `${import.meta.env.VITE_PORTAL_NAME} - ${pageName}`
+    const newTitle: string = documentTitle
+    if (document.title !== newTitle) {
+      document.title = newTitle
+    }
+  }
```

Comment:
Needed to add this code to update `document.title` because the recursive `RouteResolver` doesn't inspect these sub routes. Without this, the document title ends up becoming `"Explore :slug/"`!